src/agent/coverage/src/debuginfo.rs

```diff
@@ -76,18 +82,25 @@ impl ModuleDebugInfo {
     ///
     /// Leaks module and symbol data.
     fn load(module: &Path) -> Result<Option<Self>> {
+        // Used when `cfg(windows)`.
+        #[allow(unused_mut)]
         let mut data = fs::read(&module)?.into_boxed_slice();
 
-        // If our module is a PE file, the debug info will be in the PDB.
-        //
-        // We will need a similar check to support split DWARF.
-        let is_pe = pe::PeObject::test(&data);
-        if is_pe {
-            // Assume a sibling PDB.
+        // Conditional so we can use `dbghelp`.
+        #[cfg(windows)]
+        {
+            // If our module is a PE file, the debug info will be in the PDB.
             //
-            // TODO: Find PDB using `dbghelp::`SymGetSymbolFile()`.
-            let pdb = module.with_extension("pdb");
-            data = fs::read(&pdb)?.into_boxed_slice();
+            // We will need a similar check to support split DWARF.
+            let is_pe = pe::PeObject::test(&data);
+            if is_pe {
+                let pe = PE::parse(&data)?;
```

Comment:
This is wasteful, but unavoidable, since we need to get at the debug info header in the PE.

We use the `goblin::pe::PE` because the `PeObject` from `symbolic` does not expose enough codeview info. I opened an issue about that upstream.

Reply:
Also: since we're caching loaded debuginfo, we only ever do this once per PE module.